### ISSUE TITLE
[Baremetal IPI] startironic.sh[18246]: F1218 15:14:41.589464       1 image.go:32] error: error: Unknown name requested, could not find ironic-rhcos-downloader in UpdatePayload

### ISSUE THREAD

**Issue by mcornea** [openshift/installer#2842]
# Version

```console
$ openshift-install version
./openshift-baremetal-install v4.4.0
built from commit 459da7148535856381dd0ac4b20cbd284a7e1d37
release image registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242

nightly-2019-12-18-064553
```

# Platform:
Baremetal IPI

# What happened?

ironic systemd service fails to start on bootstrap VM:

```shell
Dec 18 15:16:29 localhost systemd[1]: Started Baremetal Deployment Ironic Services.
Dec 18 15:16:29 localhost startironic.sh[21637]: + . /usr/local/bin/release-image.sh
Dec 18 15:16:29 localhost startironic.sh[21637]: +++ podman inspect registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242 --format '{{ index .RepoDigests 0 }}'
Dec 18 15:16:29 localhost startironic.sh[21637]: ++ RELEASE_IMAGE_DIGEST=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242
Dec 18 15:16:29 localhost startironic.sh[21637]: ++ [[ -z registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242 ]]
Dec 18 15:16:29 localhost startironic.sh[21637]: ++ image_for ironic
Dec 18 15:16:29 localhost startironic.sh[21637]: ++ podman run --quiet --rm --net=none registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242 image ironic
Dec 18 15:16:29 localhost podman[21652]: 2019-12-18 15:16:29.896800358 +0000 UTC m=+0.173860799 container create d18871e7602b073de28f0860888c82300b2b71ddc0eddea9810543f867a1facf (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:30 localhost podman[21652]: 2019-12-18 15:16:30.236502084 +0000 UTC m=+0.513561923 container init d18871e7602b073de28f0860888c82300b2b71ddc0eddea9810543f867a1facf (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:30 localhost podman[21652]: 2019-12-18 15:16:30.257729532 +0000 UTC m=+0.534789416 container start d18871e7602b073de28f0860888c82300b2b71ddc0eddea9810543f867a1facf (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1>
Dec 18 15:16:30 localhost podman[21652]: 2019-12-18 15:16:30.257858565 +0000 UTC m=+0.534918321 container attach d18871e7602b073de28f0860888c82300b2b71ddc0eddea9810543f867a1facf (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:30 localhost podman[21652]: 2019-12-18 15:16:30.557350896 +0000 UTC m=+0.834410838 container died d18871e7602b073de28f0860888c82300b2b71ddc0eddea9810543f867a1facf (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:30 localhost podman[21652]: 2019-12-18 15:16:30.627553692 +0000 UTC m=+0.904613688 container remove d18871e7602b073de28f0860888c82300b2b71ddc0eddea9810543f867a1facf (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:30 localhost startironic.sh[21637]: + IRONIC_IMAGE=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:7ac28156932830e37473039958c3539763fa2570b7d7ec9182df2cfdac480d77
Dec 18 15:16:30 localhost startironic.sh[21637]: ++ image_for ironic-inspector
Dec 18 15:16:30 localhost startironic.sh[21637]: ++ podman run --quiet --rm --net=none registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242 image ironic-inspector
Dec 18 15:16:30 localhost podman[21718]: 2019-12-18 15:16:30.805772803 +0000 UTC m=+0.154460876 container create 977134862da55cfc6cd75a66f7d199980228c5311481d83f147067d10cab3671 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:31 localhost podman[21718]: 2019-12-18 15:16:31.137038097 +0000 UTC m=+0.485726234 container init 977134862da55cfc6cd75a66f7d199980228c5311481d83f147067d10cab3671 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:31 localhost podman[21718]: 2019-12-18 15:16:31.156619719 +0000 UTC m=+0.505307866 container start 977134862da55cfc6cd75a66f7d199980228c5311481d83f147067d10cab3671 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1>
Dec 18 15:16:31 localhost podman[21718]: 2019-12-18 15:16:31.156778824 +0000 UTC m=+0.505466966 container attach 977134862da55cfc6cd75a66f7d199980228c5311481d83f147067d10cab3671 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:31 localhost podman[21718]: 2019-12-18 15:16:31.486592125 +0000 UTC m=+0.835280343 container died 977134862da55cfc6cd75a66f7d199980228c5311481d83f147067d10cab3671 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:31 localhost podman[21718]: 2019-12-18 15:16:31.579909061 +0000 UTC m=+0.928597247 container remove 977134862da55cfc6cd75a66f7d199980228c5311481d83f147067d10cab3671 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:31 localhost startironic.sh[21637]: + IRONIC_INSPECTOR_IMAGE=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:93e98a790ee64d0889f699f19af236c2bb0cf08daf2c34e95c009be996397da6
Dec 18 15:16:31 localhost startironic.sh[21637]: ++ image_for ironic-ipa-downloader
Dec 18 15:16:31 localhost startironic.sh[21637]: ++ podman run --quiet --rm --net=none registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242 image ironic-ipa-downloader
Dec 18 15:16:31 localhost podman[21812]: 2019-12-18 15:16:31.799358634 +0000 UTC m=+0.187316911 container create a82740027ac68ee3a6f9a6ff59825f83c24da5f79eadcba79869d626978bb4b4 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:32 localhost podman[21812]: 2019-12-18 15:16:32.139755493 +0000 UTC m=+0.527713833 container init a82740027ac68ee3a6f9a6ff59825f83c24da5f79eadcba79869d626978bb4b4 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:32 localhost podman[21812]: 2019-12-18 15:16:32.155719373 +0000 UTC m=+0.543677631 container start a82740027ac68ee3a6f9a6ff59825f83c24da5f79eadcba79869d626978bb4b4 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1>
Dec 18 15:16:32 localhost podman[21812]: 2019-12-18 15:16:32.155865957 +0000 UTC m=+0.543824285 container attach a82740027ac68ee3a6f9a6ff59825f83c24da5f79eadcba79869d626978bb4b4 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:32 localhost podman[21812]: 2019-12-18 15:16:32.473518556 +0000 UTC m=+0.861476905 container died a82740027ac68ee3a6f9a6ff59825f83c24da5f79eadcba79869d626978bb4b4 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:32 localhost podman[21812]: 2019-12-18 15:16:32.53940773 +0000 UTC m=+0.927366119 container remove a82740027ac68ee3a6f9a6ff59825f83c24da5f79eadcba79869d626978bb4b4 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1>
Dec 18 15:16:32 localhost startironic.sh[21637]: + IPA_DOWNLOADER_IMAGE=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:66535a2697973d53749bc8c7f5957b9cc847ff969bf8f66f1c2856d374dd904b
Dec 18 15:16:32 localhost startironic.sh[21637]: ++ image_for ironic-rhcos-downloader
Dec 18 15:16:32 localhost startironic.sh[21637]: ++ podman run --quiet --rm --net=none registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b0a8fee76543e02019acae71d6100c1aa9505d8a895f2d10ef05cf220242 image ironic-rhcos-downloader
Dec 18 15:16:32 localhost podman[21886]: 2019-12-18 15:16:32.713406548 +0000 UTC m=+0.148151707 container create 5c35ee3b6b6f2fec62897033e12b06851a7145c1f2bbf39d3fe2915482e89eb7 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:33 localhost podman[21886]: 2019-12-18 15:16:33.026954416 +0000 UTC m=+0.461699702 container init 5c35ee3b6b6f2fec62897033e12b06851a7145c1f2bbf39d3fe2915482e89eb7 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:33 localhost podman[21886]: 2019-12-18 15:16:33.04572421 +0000 UTC m=+0.480469352 container start 5c35ee3b6b6f2fec62897033e12b06851a7145c1f2bbf39d3fe2915482e89eb7 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:33 localhost podman[21886]: 2019-12-18 15:16:33.045867834 +0000 UTC m=+0.480613071 container attach 5c35ee3b6b6f2fec62897033e12b06851a7145c1f2bbf39d3fe2915482e89eb7 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:33 localhost startironic.sh[21637]: F1218 15:16:33.324534       1 image.go:32] error: error: Unknown name requested, could not find ironic-rhcos-downloader in UpdatePayload
Dec 18 15:16:33 localhost podman[21886]: 2019-12-18 15:16:33.350490401 +0000 UTC m=+0.785235638 container died 5c35ee3b6b6f2fec62897033e12b06851a7145c1f2bbf39d3fe2915482e89eb7 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb1b>
Dec 18 15:16:33 localhost podman[21886]: 2019-12-18 15:16:33.411934501 +0000 UTC m=+0.846679739 container remove 5c35ee3b6b6f2fec62897033e12b06851a7145c1f2bbf39d3fe2915482e89eb7 (image=registry.svc.ci.openshift.org/ocp/release@sha256:fdb>
Dec 18 15:16:33 localhost startironic.sh[21637]: + COREOS_DOWNLOADER_IMAGE=
Dec 18 15:16:33 localhost systemd[1]: ironic.service: Main process exited, code=exited, status=255/n/a
Dec 18 15:16:33 localhost systemd[1]: ironic.service: Failed with result 'exit-code'.
Dec 18 15:16:43 localhost systemd[1]: ironic.service: Service RestartSec=10s expired, scheduling restart.
Dec 18 15:16:43 localhost systemd[1]: ironic.service: Scheduled restart job, restart counter is at 36.
Dec 18 15:16:43 localhost systemd[1]: Stopped Baremetal Deployment Ironic Services.
Dec 18 15:16:43 localhost systemd[1]: Started Baremetal Deployment Ironic Services.
```

# What you expected to happen?

on bootstrap VM ironic service starts without isses

# How to reproduce it (as minimally and precisely as possible)?

```console
Follow steps in https://github.com/openshift-kni/baremetal-deploy/blob/master/install-steps.md
```

**Comment by stbenjam** [comment]
/label platform/baremetal

This is due to https://github.com/openshift/installer/pull/2821, it should merge soon and end up in 4.4 builds. We'll then cherry-pick to 4.3

**Comment by hardys** [comment]
This was closed via #2821 and #2846
